framework/src/araui/core/ui_core.c

```diff
@@ -273,21 +285,31 @@ static ui_error_t _ui_render_widget_recur(ui_widget_body_t *widget, ui_rect_t dr
 		return UI_INVALID_PARAM;
 	}
 
-	if (widget->visible) {
-		new_vp = ui_rect_intersect(draw_area, widget->global_rect);
-		ui_dal_set_viewport(new_vp.x, new_vp.y, new_vp.width, new_vp.height);
+	ui_widget_queue_init();
+	ui_widget_queue_push(widget);
 
-		if (widget->render_cb) {
-			new_vp = ui_rect_intersect(draw_area, widget->global_rect);
+	while (!ui_widget_is_queue_empty()) {
+		curr_widget = ui_widget_queue_pop();
```

Comment:
In general, the terms 'push / pop' is used for only stack. It means LIFO.
In the queue data structure.. need to be changed to 'enqueue / dequeue'. It means FIFO.

or enq / deq